frontend/src/scalars/ui/Chart.vue

```diff
@@ -0,0 +1,598 @@
+            seriesOption = flatten(seriesOption);
+            let legendOptions = tagList.map(item => item.run);
+            let that = this;
+            let option = {
+                color: [
+                    '#c23531',
+                    '#61a0a8',
+                    '#d48265',
+                    '#91c7ae',
+                    '#749f83',
+                    '#ca8622',
+                    '#bda29a',
+                    '#6e7074',
+                    '#546570',
+                    '#c4ccd3'
+                ],
+                title: {
+                    text: tag,
+                    textStyle: {
+                        fontSize: '12'
+                    }
+                },
+                tooltip: {
+                    trigger: 'axis',
+                    axisPointer: {
+                        animation: true
+                    },
+                    position: ['10%', '90%'],
+                    formatter(params, ticket, callback) {
+                        let data = that.getFormatterPoints(params[0].data);
+                        return that.transformFormatterData(data);
+
+                    }
+                },
+                toolbox: {
+                    show: true,
+                    showTitle: true,
+                    feature: {
+                        dataZoom: {},
+                        restore: {},
+                        saveAsImage: {}
+                    },
+                    left: '10%',
+                    top: '90%'
+                },
+                legend: {
+                    data: legendOptions,
+                    top: '10%'
+                },
+                grid: {
+                    left: '10%',
+                    top: '20%',
+                    right: '10%',
+                    bottom: '20%'
+                },
+                xAxis: {
+                    type: 'value'
+                },
+                yAxis: {
+                    type: 'value',
+                    axisLabel: {
+                        formatter(value) {
+                            return value.toString().slice(0, 5);
+                        }
+                    }
+                },
+                series: seriesOption
+            };
+            this.myChart.setOption(option);
+        },
+
+        // Get origin data per 60 seconds
+        startInterval() {
+            this.getOringDataInterval = setInterval(() => {
+                this.getOriginChartData(this.tagInfo);
+            }, intervalTime * 1000);
+        },
+
+        stopInterval() {
+            clearInterval(this.getOringDataInterval);
+        },
+
+        getOriginChartData({tagList, tag}) {
+            let requestList = tagList.map(item => {
+                let params = {
+                    run: item.run,
+                    tag: tag
+                };
+                return getPluginScalarsScalars(params);
+            });
+            axios.all(requestList).then(resArray => {
```

Comment:
Requests should probably not be made from within component. Can you move this to a service / API manager outside?